Enhancement/40 add outlook location field

Philldomd opened the pull request:
# Pull Request Description

Sets the Location field to a variable that is configured in the config.js file.
This will remove the question on send that blocks, asking if the user wants to send a message without location

Fixes #40 

## Checklist

- [x] Changes are limited to a single goal (avoid scope creep)
- [x] I confirm that I have read any Contribution and Development guidelines (CONTRIBUTING and DEVELOPMENT) and are following their suggestions.
- [x] I confirm that I wrote and/or have the right to submit the contents of my Pull Request, by agreeing to the _Developer Certificate of Origin_, (adding a 'sign-off' to my commits).

Ayko1595 commented on package.json:
```diff
@@ -30,17 +30,21 @@
     "@fluentui/react": "^8.52.3",
     "core-js": "^3.9.1",
     "es6-promise": "^4.2.8",
+    "eslint": "^7.32.0",
```
I don't think we need to install eslint. Office introduces eslint and prettier through its boilerplate code

Philldomd replied:
I guess it depends on IDE, the people that developed this parts used jetbrains ide.

Philldomd replied:
I can remove and try without these npm packages.

Ayko1595 replied:
Did you try removing Eslint and tested without it?

Philldomd replied:
I think i need to remove all changes in package.json and build towards a working version for node 14. I did not think 14 was still used here.

konstantin-ay reviewed (changes requested):
Thanks again for your contribution, the feature works as expected! 😁 

But I have added some comments that I think should be adressed or discussed. Also, I was not able to run the tests locally. They seem to fail for the OfficeCallHandler.test.ts file

Philldomd commented on .gitignore:
Yes, that is a file generated by npm install.

Ayko1595 replied:
Yes I know that it is generated with npm install, but it is supposed to be checked in to the repository, as recommended by npm here: https://docs.npmjs.com/cli/v10/configuring-npm/package-lock-json 😊

Philldomd commented on src/models/Config.ts:
```diff
@@ -7,6 +7,7 @@ export const defaultFontFamily = "Arial";
 
 interface Config {
   baseUrl?: string;
+  locationString?: string;
```
Sure I missed that

Philldomd commented on jest.config.js:
Will add

Philldomd commented on __tests__/OfficeCallHandler.test.ts:
```diff
@@ -0,0 +1,84 @@
+import * as configData from "../config.json";
+import Config from "../src/models/Config"
+const AddinOfficeMockObject = require("office-addin-mock");
+import { setLocationTest, setDataTest, addMeeting } from '../src/utils/OfficeCallHandler'
```
That is true :)

Philldomd commented on package-lock.json:
I can add it back for that purpose. But it had a lot of changes

Ayko1595 replied:
The package-lock.json will get a lot of changes whenever we add new dependencies, so it it expected 😊

Philldomd commented on LICENSES/CC0-1.0.txt:
They are changed when downloaded to windows. I missed to remove this commit here, I will push a new branch without this permission change

Ayko1595 replied:
Couldn't you just fix them as part of a commit on the same branch?

Philldomd replied:
I left it in its own commit so that i could preserve it on upload here. That commit  should never have been pushed.

Philldomd commented on tsconfig.json:
```diff
@@ -5,7 +5,7 @@
     "esModuleInterop": true,
     "experimentalDecorators": true,
     "jsx": "react",
-    "module": "CommonJS",
```
Hmm, no I think this snuck in here when I was looking for a bug

Philldomd commented:
> Thanks again for your contribution, the feature works as expected! 😁
> 
> But I have added some comments that I think should be adressed or discussed. Also, I was not able to run the tests locally. They seem to fail for the OfficeCallHandler.test.ts file

What problem did you get here? Was npm install run to get latest changes in package?

Ayko1595 commented:
@Philldomd Yes I have the latest version of the code and also ran npm install. Here is the error message I received:

![Screenshot 2024-06-11 at 13 13 14](https://github.com/diggsweden/jitsi-outlook/assets/14002128/cfd0a1be-e32f-4dd9-a771-5fb4ee19ac63)

Philldomd commented:
@Ayko1595 If you change to the following will that work?
10 import { OfficeMockObject } from "office-addin-mock";
35 const setLocationMock = new OfficeMockObject(mockSetLocationServer) as any;
69 const setDataMock = new OfficeMockObject(mockSetDataServer) as any;

does the test work then?

Ayko1595 commented:
@Philldomd It does not work after those changes, I am still getting those errors. I think the issue could be the testing environment. The jest configuration specifies a jsdom environment, which is used on client-side testing. The tests you want to run in the OfficeCallHandler relies on a node environment which causes the test to fail. There is probably a mismatch in the configurations and the test files

Philldomd commented:
I changed to Node instead and all other tests failed. I get everything to pass, both on my workstation and my laptop. So i can't reproduce this. I have removed package-lock and node_modules and run install again with the same resault. Have you added office-addin-mock?

Ayko1595 commented:
Office-addin-mock is installed yes. I am running this on MacOS and it still fails unfortunately. If you fix the comment regarding the Jest config's license header we will be able to run the GitHub action that tests the code

Ayko1595 commented on .github/workflows/workflow.yml:
```diff
@@ -17,4 +17,4 @@ jobs:
     uses: ./.github/workflows/license.yml
   test:
     needs: [license]
-    uses: ./.github/workflows/tests.yml
```
Hmm I think GitHub is drunk? I can't see any change here

janderssonse commented:
Off-topic: Nice to see you at the meeting @Philldomd and while have 0% time left this week to help out reviewing, I'm very glad to see these PR's arrive to the project, and that FK is part of this. Much cred to all involved in this. This is collaborationn for real.

Philldomd committed:
40: Add location to outlook location field

Also added some more testcases.

Signed-off-by: Philip Örnfeldt <philip.ornfeldt@outlook.com>

Philldomd commented:
Code is squashed and updated as requested

Philldomd commented:
Testing with older package.json file

Ayko1595 commented on package.json:
```diff
@@ -81,10 +83,23 @@
     "tabWidth": 2,
     "printWidth": 200
   },
+  "eslintConfig": {
```
Just a minor thing: 
Do we need to have this eslintConfig in the package.json anymore?

Philldomd replied:
I think that was added for an IDE plugin to work with eslint. Not sure, removed it for now

konstantin-ay reviewed (approved):
Looks good! 😄🤝

Philldomd commented:
I found out now from a co-worker that this actually introduces a bug, the location field is cleared from other information upon generating a jitsi link. I will have a look into i and ensure that this does not happen here. Await my next commit for that fix and a testcase that checks it.

Ayko1595 commented:
@Philldomd Oh okay, well nice catch by our co-worker. If it is possible to introduce a unit test that tests this issue it would be great :-)

konstantin-ay reviewed (changes requested):
Bug was found so I change this status to "Request changes"

Philldomd commented:
Okey now I think this one is finally finished. Added a lot of tests to control the location field. At least it is not duplicating the text or removing other location field information.

Ayko1595 commented:
Looks good @Philldomd but there seems to be a conflict in the commands.ts file

Philldomd commented:
Let's wait for bugfix and then merge this code in. There will be some major changes when that happens

Philldomd committed:
Update OfficeCallHandler.ts

Signed-off-by: Philip Örnfeldt <1583206+Philldomd@users.noreply.github.com>

Philldomd committed:
Merge branch 'main' into enhancement/40_Add_Outlook_location_field

Signed-off-by: Philip Örnfeldt <1583206+Philldomd@users.noreply.github.com>
